### PR TITLE
"Button" component - Add "tertiary" color variant

### PR DESCRIPTION
**Pull request by didoo**
## :pushpin: Summary

Recently the designers have added a `tertiary` color variant to the `Button` component (see [Figma file](https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/?node-id=2%3A9)). 

We have to port the same changes in code.

## :hammer_and_wrench: Detailed description

In this PR I have:
- added the `tertiary` color variant to the `Button` component (JS + CSS)
- updated the documentation page for the `Button` component
   - added the `tertiary` variant to the "how to" and showcase
   - visually split the “status” grid for better readability
   - updated the design guidelines image
 
_Notice: while working on this, I notice that the way we're applying the "elevation" `box-shadow` to the buttons is not correct. I'll open a follow-up PR to fix this._

Preview of the documentation page: https://hds-components-git-add-tertiary-button-hashicorp.vercel.app/components/button

## :camera_flash: Screenshots

screenshot_1098

***

##  👀 How to review

👉 Review commit-by-commit or by files changed (as you prefer)

Reviewer's checklist:
- [ ] +1 Percy
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201977868744730

### CODE REVIEW

File: packages/components/tests/dummy/app/templates/components/button.hbs [hunk 1]
```diff
@@ -111,6 +112,9 @@
   {{! prettier-ignore-end }}
   <p class="dummy-paragraph">Renders to:</p>
   <Hds::Button @text="Copy to clipboard" @icon="clipboard-copy" />
+  <p class="dummy-paragraph"><strong>🚨 IMPORTANT:</strong>
+    <code class="dummy-code">tertiary</code>
+    buttons are required to have either a leading or trailing icon to be accessible.</p>
```
Comment (amyrlam) [inline packages/components/tests/dummy/app/templates/components/button.hbs:117]: Question: Code looks good to me. Should we add an icon to the stickersheet before merge to match ^? Also in the stickersheet "States"

image

Comment (didoo) [reply packages/components/tests/dummy/app/templates/components/button.hbs:117]: This is something Josh Klekamp will fix in the documentation and sticketsheet in Figma. Once done I'll update (again) the image.

**Review by heatherlarsen — APPROVED** [review]
This looks good! I'll get you an updated screenshot of the docs.

File: packages/components/tests/dummy/app/templates/components/button.hbs (hunk 1 above)
Comment (didoo) [reply packages/components/tests/dummy/app/templates/components/button.hbs:117]: @amyrlam https://github.com/hashicorp/design-system/pull/90

**Pull request merged**